### PR TITLE
Add SubQueue option and remove locktoken overloads

### LINKED ISSUE #14490: Use ReceiverOptions instead of overload for CreateDeadLetterReceiver
Will need a new enum for `SubQueueType`: None, DeadLetter, TransferDeadLetter.

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/14490 and https://github.com/Azure/azure-sdk-for-net/issues/14493

### CODE REVIEW

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/SubQueue.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        /// <item>
+        /// <description>A message passes through more than four queues or topics that are chained together.</description>
+        /// </item>
+        /// <item>
+        /// <description>The destination queue or topic is disabled or deleted.</description>
+        /// </item>
+        /// <item>
+        /// <description>The destination queue or topic exceeds the maximum entity size.</description>
+        /// </item>
+        /// </list>
```
Comment (MiYanni) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/SubQueue.cs:35]: Something with the XML docs went crazy here.

**Review by MiYanni — APPROVED** [review]

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/SubQueue.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/SubQueue.cs:35]: What do you mean? This is to show a bulleted list.

Comment (MiYanni) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/SubQueue.cs:35]: Interesting! I've never seen that before.

**Pull request merged**